lib/UserSettings.php

```diff
@@ -154,9 +154,13 @@ protected function getDefaultFromSetting($method, $type) {
 	 */
 	public function getNotificationTypes() {
 		$settings = $this->manager->getSettings();
-		return array_map(function (ActivitySettings $setting) {
+		
+		$ret = array();
+		$ret = array_map(function (ActivitySettings $setting) {
 			return $setting->getIdentifier();
 		}, $settings);
+		array_push($ret, 'file_created', 'file_deleted', 'file_restored');
+		return $ret;
```

Comment:
```suggestion
		$return = array_map(function (ActivitySettings $setting) {
			return $setting->getIdentifier();
		}, $settings);
		array_push($return, 'file_created', 'file_deleted', 'file_restored');
		return $return;
```

Also check for `file_changed` before adding them with https://www.php.net/manual/en/function.array-search.php

Reply:
Thanks, fixed it!